reporter/functional-tests/src/test/value-class-added-concrete-method-nok/problems.txt

```diff
@@ -0,0 +1 @@
+extension method c$extension(java.lang.String,Int)Unit in object A#B does not have a correspondent in new version
```

Comment:
No, I don't think so. Adding additional overloads should be fine but not adding a first overload.

The `$extension` methods get an appended number for disambiguation (see https://github.com/scala/scala/blob/9c5d3f81a667917b6a3aed5098182623c417c137/src/compiler/scala/tools/nsc/transform/ExtensionMethods.scala#L30-L38). It is not clear to me from that comment why we need to do this. The signature is the same as the one of the trait method, so I would expect that we can safely overload it in the same way.

Even if we cannot, perhaps we should use `$extension0` instead of `$extension` for the non-overloaded case?